astropy/utils/decorators.py

```diff
@@ -496,6 +510,8 @@ def wrapper(*args, **kwargs):
                                 raise TypeError(
                                     'cannot specify both "{}" and "{}"'
                                     '.'.format(old_name[i], new_name[i]))
+                    elif new_name[i] is None:
```

Comment:
I would combine this with the `else` or move the comment (the "# If the new argument isn't specified..."), otherwise this will be a bit hard to follow.